### PR TITLE
refactor: append, build variable and type switch

### PR DESCRIPTION
**Pull request by appleboy**
1. `append` chains to the same slice that can be done in a single `append` call
2. when predeclared identifiers shadowed in assignments
3. loops that copy big objects during each iteration
4. type switches that can benefit from type guard clause with a variable

### CODE REVIEW

**Comment by lafriks** [comment]
Blocked till 1.6.0-rc1 is released

File: models/user.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"errors"
 	"fmt"
 	"image"
+
```
Comment (techknowlogick) [inline models/user.go]: Why space?

Comment (appleboy) [reply models/user.go]: See https://github.com/go-gitea/gitea/pull/4940/commits/a4f370b90cd2301b4933759f52bccf08b7fad1bb

**Comment by lafriks** [comment]
Please resolve conflicts and sorry for forgetting to review it

**Comment by lafriks** [comment]
@appleboy conflicts

**Comment by appleboy** [comment]
@lafriks resolved conflicts.

File: modules/base/tool.go [hunk 2]
```diff
@@ -465,41 +465,41 @@ func Subtract(left interface{}, right interface{}) interface{} {
...
+		fleft = float64(v)
 		isInt = false
 	case float64:
-		fleft = left
+		fleft = v
 		isInt = false
 	}
 
-	switch right := right.(type) {
+	switch v := right.(type) {
```
Comment (zeripath) [inline modules/base/tool.go:487]: Could we not refactor this further to get rid of the code duplication completely?

**Review by zeripath — APPROVED** [review]

File: models/issue_assignees_test.go [hunk 3]
```diff
@@ -43,8 +43,7 @@ func TestUpdateAssignee(t *testing.T) {
 	assert.NoError(t, err)
 
 	var expectedAssignees []*User
-	expectedAssignees = append(expectedAssignees, user2)
-	expectedAssignees = append(expectedAssignees, user3)
+	expectedAssignees = append(expectedAssignees, user2, user3)
```
Comment (lunny) [inline models/issue_assignees_test.go:46]: ~I think this may changed the tests?~

**Review by lunny — CHANGES_REQUESTED** [review]

**Review by sapk — APPROVED** [review]

File: models/issue_assignees_test.go (hunk 3 above)
Comment (appleboy) [reply models/issue_assignees_test.go:46]: @lunny I don't know what do you mean?

**Review by lunny — APPROVED** [review]

**Pull request merged**